src/charm.py

```diff
@@ -32,140 +31,144 @@
     MetricsEndpointConsumer,
     MetricsEndpointProvider,
 )
-from charms.traefik_k8s.v1.ingress_per_unit import IngressPerUnitRequirer
+from charms.traefik_k8s.v1.ingress_per_unit import (
+    IngressPerUnitReadyForUnitEvent,
+    IngressPerUnitRequirer,
+    IngressPerUnitRevokedForUnitEvent,
+)
 from lightkube import Client
 from lightkube.core.exceptions import ApiError as LightkubeApiError
 from lightkube.resources.core_v1 import PersistentVolumeClaim, Pod
-from lightkube.utils.quantity import parse_quantity
 from ops.charm import ActionEvent, CharmBase
 from ops.main import main
 from ops.model import ActiveStatus, BlockedStatus, MaintenanceStatus, WaitingStatus
 from ops.pebble import ChangeError, ExecError, Layer
 
+from conversion import convert_k8s_quantity_to_legacy_binary_gigabytes
```

Comment:
`from utils import`?

Reply:
Going with `utils` for now. I don't think it's more controversial than just not wanting it to be in the spotlight of the file when it is so disconnected from the rest of the code. It also removed some imports from the charm file as they instead went with the function to the utils file.